--- a/src/melee/ft/fighter.h
+++ b/src/melee/ft/fighter.h
@@ -172,4 +172,8 @@ extern struct Fighter_804D64FC_t {
 /* 4D6530 */ extern Vec2** Fighter_804D6530;
-/* 4D6534 */ extern UNK_T Fighter_804D6534;
+// refs/melee/src/melee/ft/ft_0D4D.c::ftCo_800D4FF4
+/* 4D6534 */ extern struct Fighter_804D6534_t {
+    HSD_Joint* joint;
+    HSD_AnimJoint* anim_joint;
+}* Fighter_804D6534;
 /* 4D6538 */ extern struct Fighter_804D653C_t* Fighter_804D6538;
 /* 4D653C */ extern struct Fighter_804D653C_t* Fighter_804D653C;
--- a/src/melee/ft/fighter.c
+++ b/src/melee/ft/fighter.c
@@ -145,5 +145,5 @@ struct Fighter_804D6528_t* Fighter_804D6528 = NULL;
 UNK_T Fighter_804D652C = NULL;
 Vec2** Fighter_804D6530 = NULL;
-UNK_T Fighter_804D6534 = NULL;
+struct Fighter_804D6534_t* Fighter_804D6534 = NULL;
 struct Fighter_804D653C_t* Fighter_804D6538 = NULL;
 struct Fighter_804D653C_t* Fighter_804D653C = NULL;
--- a/src/melee/ft/ft_0D4D.c
+++ b/src/melee/ft/ft_0D4D.c
@@ -139,4 +139,4 @@ void ftCo_800D4FF4(Fighter_GObj* gobj)
 
     if (!fp->x221F_b4) {
-        ftCommon_SetAccessory(fp, ((void**) Fighter_804D6534)[0]);
+        ftCommon_SetAccessory(fp, Fighter_804D6534->joint);
         {
             f32 s = fp->x34_scale.y * fp->co_attrs.respawn_platform_scale;
@@ -148,5 +148,5 @@ void ftCo_800D4FF4(Fighter_GObj* gobj)
         HSD_JObjSetScale(fp->x20A0_accessory, &scale_vec);
 
-        ftCommon_8007E690(fp, ((void**) Fighter_804D6534)[1]);
+        ftCommon_8007E690(fp, Fighter_804D6534->anim_joint);
 
         pos_vec.x = -(fp->facing_dir * ftCommon_800804EC(fp) - fp->cur_pos.x);
         pos_vec.y = fp->cur_pos.y;
